ml-agents/mlagents/trainers/tests/test_simple_rl.py

```diff
@@ -262,28 +182,45 @@ def _check_environment_trains(
 
         # Begin training
         tc.start_learning(env_manager)
-        print(tc._get_measure_vals())
         if (
             success_threshold is not None
         ):  # For tests where we are just checking setup and not reward
-            for mean_reward in tc._get_measure_vals().values():
-                assert not math.isnan(mean_reward)
-                assert mean_reward > success_threshold
+
+            processed_rewards = [
+                reward_processor(rewards) for rewards in env.final_rewards.values()
+            ]
+            assert all(not math.isnan(reward) for reward in processed_rewards)
+            if not check_fail:
+                assert all(reward > success_threshold for reward in processed_rewards)
+            else:
+                assert any(
+                    reward > success_threshold for reward in processed_rewards
+                ) and any(reward < success_threshold for reward in processed_rewards)
 
 
 @pytest.mark.parametrize("use_discrete", [True, False])
 def test_simple_ppo(use_discrete):
-    env = Simple1DEnvironment(use_discrete=use_discrete)
+    env = Simple1DEnvironment([BRAIN_NAME], use_discrete=use_discrete)
     _check_environment_trains(env, PPO_CONFIG)
 
 
 @pytest.mark.parametrize("use_discrete", [True, False])
 def test_simple_sac(use_discrete):
-    env = Simple1DEnvironment(use_discrete=use_discrete)
+    env = Simple1DEnvironment([BRAIN_NAME], use_discrete=use_discrete)
     _check_environment_trains(env, SAC_CONFIG)
 
 
 @pytest.mark.parametrize("use_discrete", [True, False])
 def test_simple_ghost(use_discrete):
-    env = Simple1DEnvironment(use_discrete=use_discrete)
-    _check_environment_trains(env, GHOST_CONFIG)
+    env = Simple1DEnvironment(
```

Comment:
Maybe move these (and GHOST_CONFIG_*) to their own file?

Reply:
@andrewcoh should we just move these tests to their respective places? e.g. `test_ghost.py`, `test_ppo.py`, etc.

Reply:
I could move this to test_ghost.py. Then, it also might make sense to move the ppo/sac stuff to test_ppo.py and test_sac.py and leave this file as just _check_env_trains.  Thoughts @ervteng ?

Reply:
I should've refreshed :)